frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.tsx

```diff
@@ -242,83 +246,110 @@ export function FilterPopover({
   const supportsExpressions = query.database()?.supportsExpressions();
 
   const filterOperator = filter.operator();
-  const hasPicker = filterOperator && filterOperator.fields.length > 0;
+  const hasPicker =
+    (filterOperator && filterOperator.fields.length > 0) || isBoolean(field);
+  const stageIndex = -1; // this will not always be true
+
+  const {
+    filterClause,
+    query: mlv2Query,
+    column,
+    // TODO: don't blow up trying to make an MLv2 Date filter
+  } = isDate(dimension)
+    ? filter.getMlv2FilterClause({ stageIndex })
+    : {
+        filterClause: undefined,
+        query: undefined,
+        column: undefined,
+      };
 
   return (
     <div className={className} style={{ minWidth: MIN_WIDTH, ...style }}>
-      {shouldShowDatePicker ? (
-        <DatePicker
-          className={className}
-          filter={filter}
-          dateShortcutOptions={dateShortcutOptions}
-          primaryColor={primaryColor}
-          minWidth={MIN_WIDTH}
-          maxWidth={MAX_WIDTH}
-          onBack={onBack}
-          onCommit={handleCommit}
-          onFilterChange={handleFilterChange}
-          disableChangingDimension={!showFieldPicker}
-          supportsExpressions={supportsExpressions}
+      <ErrorBoundary>
+        <FilterContext.Provider
+          value={{
+            filter: filterClause,
+            query: mlv2Query,
+            legacyQuery: query,
+            column: column,
+            stageIndex,
+          }}
```

Comment:
I think it's worth introducing a `FilterContextProvider` component that'd take an MLv1 filter and do all the conversation heavy lifting internally. We should also be able to guard ourselves from null-ish queries in there

Reply:
yeah, I think using context was just slightly too clever here. take a look at the version without context and see what you think.